solr/core/src/java/org/apache/solr/core/FileSystemConfigSetService.java

```diff
@@ -287,11 +285,9 @@ public Long getCurrentSchemaModificationVersion(
   }
 
   protected Path getConfigDir(String configName) throws IOException {
-    String path =
-        Paths.get(configSetBase.toString(), FilenameUtils.getName(configName))
-            .toFile()
-            .getCanonicalPath();
-    if (!path.endsWith(configName)) {
+    String configSetBasePath = configSetBase.toString();
+    String path = Path.of(configSetBasePath, configName).toFile().getCanonicalPath();
+    if (!path.startsWith(configSetBasePath)) {
```

Comment:
No, I meant Path.startsWith(path).  In general, hold paths as Path not String nor File.  It's typed, has lots of useful methods, and easier to "get right" whereas String processing in particular tends to have more subtle bugs.